server/core/config/raw/persistence.go

```diff
@@ -0,0 +1,89 @@
+package raw
+
+import (
+	validation "github.com/go-ozzo/ozzo-validation"
+	"github.com/graymeta/stow"
+	stow_s3 "github.com/graymeta/stow/s3"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+)
+
+type Persistence struct {
+	// DefaultStore is the name of the default data store to use
+	DefaultStore string `yaml:"default_store" json:"default_store"`
+	// DataStores contains the configuration for all datastores
+	DataStores DataStores `yaml:"data_stores" json:"data_stores"`
+	// Adds a prefix to the storage container
+	Prefix string `yaml:"prefix" json:"prefix"`
+
+	DeploymentStore string `yaml:"deployment_store" json:"deployment_store"`
+	JobStore        string `yaml:"job_store" json:"job_store"`
```

Comment:
You mean `JobStore` and `DeploymentStore`? We can either chose to configure the default store which is used for for both deployments and jobs, or choose to override the default with a different data-store. 

I have added a comment here for reference. Our config would look like 
```
persistence:
  default_store: default
  prefix: atlantis-deploy-artifacts
  data_stores:
    default:
      s3:
        bucket-name: atlantis-staging-states
```

If prefix = "deploy-job-artifacts", the keys will be
```
- deployment-keys = deploy-job-artifacts/deployments/repo/root/deployment.json
- job_keys = /deploy-job-artifacts/jobs/ouptut/uuid
```

Reply:
1. When i mentioned modeling it similar i didn't mean copying it directly, I'm not sure why we need default_store and the list of datastores, is there a good reason for that? I'm curious
2. seems like you're using the same prefix for deploy and jobs so lets just make sure we're not changing any existing job output locations since it'll affect looking up prior logs.

Reply:
Ahh my bad, I figured that was the standard approach of configuring persistence and that we wanted something similar. Having a default store and different data stores allows us to configure different backends for jobs and deployments but I guess we don't need that since we're only using s3. 

Regarding looking up prior logs, since we're only using this setup for Neptune, it should not affect looking up prior logs.

Reply:
As discussed offline, we should be using the same bucket for everything regardless of neptune.